spec/spec_helper.rb

```diff
@@ -195,22 +199,60 @@ def record(prefix, bindings, explicit_cassette_name, example: nil)
       yield if block_given?
     end
 
-    if ENV['VCR_UPDATE'] == '1'
+    record_cassette(prefix, bindings, explicit_cassette_name, run)
+  end
+
+  def record_websocket(prefix, bindings, explicit_cassette_name, example)
+    prefix.gsub!(/\s/, '_')
+    explicit_cassette_name.gsub!(/\s/, '_')
+    run = lambda do
+      unless example.nil?
+        if example.respond_to?(:run)
+          example.run
+        elsif example.respond_to?(:call)
+          example.call
+        end
+      end
+      yield if block_given?
+    end
+
+    record_websocket_cassette(prefix, bindings, explicit_cassette_name, run)
+  end
+
+  private
+
+  def record_cassette(prefix, bindings, explicit_cassette_name, run_lambda)
+    if ENV['VCR_UPDATE'] == '1' && bindings
       VCR.use_cassette(prefix + '/tmp/' + explicit_cassette_name,
                        decode_compressed_response: true,
                        record: :all) do
-        run.call
+        run_lambda.call
       end
       make_template prefix, explicit_cassette_name, bindings
     else
       VCR.use_cassette(prefix + '/Templates/' + explicit_cassette_name,
                        decode_compressed_response: true,
                        erb: bindings,
-                       record: :none) do
-        run.call
+                       record: ENV['VCR_UPDATE'] == '1' ? :all : :none) do
+        run_lambda.call
       end
     end
   end
+
+  def record_websocket_cassette(prefix, bindings, explicit_cassette_name, run_lambda)
+    options = {
+      record: :none,
+      decode_compressed_response: true
+    }
+    options[:erb] = bindings if bindings
+    if ENV['VCR_UPDATE'] == '1'
+      options[:record] = :all
+      options[:reverse_substitution] = true if bindings
+    end
+    WebSocketVCR.use_cassette(prefix + '/' + explicit_cassette_name, options) do
+      run_lambda.call
+    end
+  end
```

Comment:
lgtm, perhaps [this](https://github.com/josejulio/hawkular-client-ruby/blob/ecda95aa1e6f48eba99ee5c48e7c101b0a621566/spec/integration/inventory_spec.rb#L493) is worth rewriting with your helper function, but it shouldn't block this PR